dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -2048,6 +2048,11 @@ public void archive(final Contentlet contentlet, final User user, final boolean
                 new ContentletLoader().invalidate(contentlet);
 
                 publishRelatedHtmlPages(contentlet);
+
+                if (contentlet.isHTMLPage()) {
+                    CacheLocator.getHTMLPageCache().remove(contentlet.getInode());
+                }
```

Comment:
I think  you need to call 
`new PageLoader().invalidate(page)` as well

Reply:
maybe we add this to the HTMLPageCacheImpl whenever a page is invalidated

Reply:
ready